### PR TITLE
Improvement of Red Black Tree

### PR DESCRIPTION
**Pull request by remlostime**
### Checklist

- [x] I've looked at the [contribution guidelines](https://github.com/raywenderlich/swift-algorithm-club/blob/master/.github/CONTRIBUTING.md).
- [x] This pull request is complete and ready for review.

### Description
* Add count(), allElements(), isEmpty() functions
* Make the config right for RBTree in workspace
* Create a separate class for RBTree

### CODE REVIEW

**Comment by JaapWijnen** [comment]
Looking good! Solid improvements. There are some conflicts though, could you resolve these?

**Comment by remlostime** [comment]
Sure. Will do it. @JaapWijnen

**Comment by remlostime** [comment]
Rebased

File: Red-Black Tree/RedBlackTree.playground/Sources/RedBlackTree.swift [hunk 1]
```diff
@@ -21,730 +21,775 @@
 import Foundation
 
 private enum RBTreeColor {
-  case red
```
Comment (JaapWijnen) [inline Red-Black Tree/RedBlackTree.playground/Sources/RedBlackTree.swift]: I still get changes in spacing like this

**Comment by remlostime** [comment]
@JaapWijnen Sorry for inconvenience. Update the indent.

**Comment by JaapWijnen** [comment]
Sorry for taking so long! Just one more addition and then it's good to go I think. Could you also mention the added functions in the readme?

**Comment by remlostime** [comment]
@JaapWijnen Updated readme.

**Comment by remlostime** [comment]
@JaapWijnen @kelvinlauKL PTAL

**Comment by remlostime** [comment]
@JaapWijnen Could you take another look?

**Comment by kelvinlauKL** [comment]
Hey @remlostime. Sorry for the long wait for a response. 

We've updated our guidelines to focus submissions to the Swift Algorithm Club on general purpose algorithms & data structures. Unfortunately, this PR is an interview-style submission, and so we are closing it as it does not fall under the general category.

**Comment by kelvinlauKL** [comment]
Closed this prematurely 😄 

Thanks for this submission @remlostime - I'm merging this in!

**Pull request merged**